From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lemures Lemniscati <lemures.lemniscati@gmail.com>
Date: Sat, 7 Nov 2020 11:42:52 +0900
Subject: [PATCH] Set SOVERSION for libfontforge when BUILD_SHARED_LIBS

At least, this is useful for cygwin build.
And expected to be a non-breaking change.

diff --git a/fontforge/CMakeLists.txt b/fontforge/CMakeLists.txt
index 51a19f427..31e77f93f 100644
--- a/fontforge/CMakeLists.txt
+++ b/fontforge/CMakeLists.txt
@@ -216,8 +216,10 @@ endif()
 # Turn up warnings on a few source files
 set_property(SOURCE splineoverlap.c APPEND PROPERTY COMPILE_OPTIONS ${FONTFORGE_EXTRA_CFLAGS})
 
-set_property(TARGET fontforge PROPERTY VERSION 4)
+set(LIBFONTFORGE_ABI 4)
+set_property(TARGET fontforge PROPERTY VERSION ${LIBFONTFORGE_ABI})
 if(BUILD_SHARED_LIBS)
+  set_property(TARGET fontforge PROPERTY SOVERSION ${LIBFONTFORGE_ABI})
   if(WIN32 OR CYGWIN)
     target_link_options(fontforge PRIVATE -Wl,--export-all-symbols) # FIXME
   endif()
